include/zio/zio_fifo.h

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_ZIO_FIFO_H_
+#define ZEPHYR_ZIO_FIFO_H_
+
+#include <zephyr/types.h>
+
+/**
+ * @private
+ * @brief Common fifo attributes
+ */
+struct zio_fifo {
+	volatile u32_t in;
+	volatile u32_t out;
+	const u32_t mask;
+	const u8_t elem_size;
+	u8_t *data;
+};
+
+/**
+ * @private
+ * @brief 2^N using bitshift
+ */
+#define z_zio_fifo_pow2(n) (1u << n)
+
+/**
+ * @brief Statically initialize a zio_fifo
+ */
+#define zio_fifo_static_init(name, type, pow)	       \
```

Comment:
Zephyr uses UPPERCASE in static initializers.

Reply:
Note taken, will adjust macros accordingly